core/src/main/java/com/linecorp/armeria/server/ExceptionReportingServerErrorHandler.java

```diff
@@ -0,0 +1,167 @@
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server;
+
+import java.time.Duration;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicIntegerFieldUpdater;
+import java.util.concurrent.atomic.LongAdder;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.linecorp.armeria.common.AggregatedHttpResponse;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.RequestHeaders;
+import com.linecorp.armeria.common.annotation.Nullable;
+import com.linecorp.armeria.common.metric.MeterIdPrefix;
+import com.linecorp.armeria.common.util.TextFormatter;
+import com.linecorp.armeria.internal.common.metric.MicrometerUtil;
+import com.linecorp.armeria.server.logging.LoggingService;
+
+import io.micrometer.core.instrument.MeterRegistry;
+
+/**
+ * A {@link ServerErrorHandler} that wraps another {@link ServerErrorHandler}
+ * to periodically report the exceptions that were not logged by decorators such as
+ * {@link LoggingService}.
+ */
+final class ExceptionReportingServerErrorHandler implements ServerErrorHandler, ServerListener {
+
+    private static final Logger logger = LoggerFactory.getLogger(ExceptionReportingServerErrorHandler.class);
+    private static final AtomicIntegerFieldUpdater<ExceptionReportingServerErrorHandler> scheduledUpdater =
+            AtomicIntegerFieldUpdater.newUpdater(ExceptionReportingServerErrorHandler.class,
+                                                 "scheduled");
+
+    private final Duration interval;
+    private final ServerErrorHandler delegate;
+    private final MicrometerCounter micrometerCounter;
+    private final LongAdder counter;
+    private long lastExceptionsCount;
+    private volatile int scheduled;
+    @Nullable
+    private Throwable thrownException;
+    @Nullable
+    private ScheduledFuture<?> reportingTaskFuture;
+
+    ExceptionReportingServerErrorHandler(MeterRegistry meterRegistry, ServerErrorHandler serverErrorHandler,
+                                         Duration interval) {
+        assert !interval.isNegative() && !interval.isZero();
+        this.interval = interval;
+        delegate = serverErrorHandler;
+        micrometerCounter =
+                MicrometerUtil.register(meterRegistry,
+                                        new MeterIdPrefix("armeria.server.exceptions.unhandled"),
+                                        MicrometerCounter.class, MicrometerCounter::new);
```

Comment:
`MicrometerUtil.register` is only useful when we want to reuse the container object that contains Micrometer meter instances. In this PR's case, we only have one Micrometer Counter instance so we don't need any container type, like we needed RequestMetrics.

Reply:
Thank you for the detailed explanation. I'll remove the use of `MicrometerUtil` and use `meterRegistry.counter(...)`.